flink-runtime/src/main/java/org/apache/flink/runtime/execution/Environment.java

```diff
@@ -119,7 +120,14 @@ public interface Environment {
     Configuration getJobConfiguration();
 
     /**
-     * Returns the {@link TaskInfo} object associated with this subtask
+     * Returns the {@link JobInfo} object associated with current job.
+     *
+     * @return JobInfo for current job
+     */
+    JobInfo getJobInfo();
```

Comment:
This does not match the commit message "Unify the provision of metadata in RuntimeContext"

Reply:
These code changes are prerequisite for the FLIP-380. I've moved these code changes to the commit `[FLINK-33905][core] Add getJobInfo() method to Environment`.